core/src/main/kotlin/transformers/documentation/PreMergeDocumentableTransformer.kt

```diff
@@ -28,8 +28,8 @@ fun PreMergeDocumentableTransformer.sourceSet(documentable: Documentable): Dokka
 fun PreMergeDocumentableTransformer.perPackageOptions(documentable: Documentable): PackageOptions? {
     val packageName = documentable.dri.packageName ?: return null
     return sourceSet(documentable).perPackageOptions
-        .sortedByDescending { packageOptions -> packageOptions.prefix.length }
-        .firstOrNull { packageOptions -> packageName.startsWith(packageOptions.prefix) }
+        .sortedByDescending { packageOptions -> packageOptions.matchingRegex.length }
```

Comment:
This feels a bit awkward with regexes. I think I would prefer the order of the rules in the Gradle file to determine priority but I'll let people more familiar with the subject decide.

Reply:
@kamildoleglo ?

Reply:
I think ordering in Gradle might be more problematic and potentially could break easier. Also it will cause some confusion especially when having split logic for dokka configuration (eg. some packageOptions in buildSrc and some in Gradle build files), so I'd personally stick with the regex for now

Reply:
Fair point about split logic, especially since it's hard to know in advance the order of execution of Gradle scripts. I added a [small note](https://github.com/Kotlin/dokka/pull/598/commits/f97135b49ae5008c05b33f690d81e2d565f43c42) to the Gradle documentation to tell users that the longuest matchingRegexp will be used. 

For the `cli` usage, maybe the order of arguments would still make sense? Or ultimately, maybe an explicit `priority` option will be needed. But this can be all be added later depending the needs and use cases